.github/workflows/update-deps.yml

```diff
@@ -0,0 +1,90 @@
+name: Update NPM dependencies
+
+on:
+  schedule:
+  - cron: "0 5 * * 1"
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        deps: [production, dev-only]
+
+    steps:
+      - name: Setup node
+        # https://github.com/marketplace/actions/setup-node-js-environment
+        uses: actions/setup-node@v1
+        with:
+          node-version: '12'
+
+      - name: Fetch changes
+        # https://github.com/marketplace/actions/checkout
+        uses: actions/checkout@v2
+
+      - name: Check env variables
+        # Expected secrets variables:
+        # BOT_USERNAME - GH user name which will be used to push changes.
+        # BOT_EMAIL - GH user email which will be used to push changes.
+        # [BRANCH_MAIN] - Target branch name (optional), if not set default branch will be used.
+        run: |
+          if [[ -z "${{ secrets.BOT_USERNAME }}" ]]; then
+            echo "Expected 'BOT_USERNAME' secret variable to be set. Current value '${{ secrets.BOT_USERNAME }}'"
+            exit 1
+          fi
+          if [[ -z "${{ secrets.BOT_EMAIL }}" ]]; then
+            echo "Expected 'BOT_EMAIL' secret variable to be set. Current value '${{ secrets.BOT_EMAIL }}'"
+            exit 1
+          fi
+          if [[ -z "${{ secrets.BRANCH_MAIN }}" ]]; then
+            # Since 'Fetch changes' step fetches default branch, we just get branch name (which will be default one).
+            echo "BRANCH_MAIN=$(git rev-parse --abbrev-ref HEAD)" >> $GITHUB_ENV
+          else
+            echo "BRANCH_MAIN=${{ secrets.BRANCH_MAIN }}" >> $GITHUB_ENV
+          fi
+
+      - name: Check if update branch already exists
+        run: |
+          echo "BRANCH_UPDATE=deps-update_${{ env.BRANCH_MAIN }}_${{ matrix.deps }}" >> $GITHUB_ENV
```

Comment:
Well, yes, I used it for tests too. For me, the use case is that you may have update deps PRs targeting e.g. `master` and then you change `BRANCH_MAIN` via GH settings. Without target branch in deps update branch name it won't be created as such branches already exist. So it introduce more flexibility and also the behavior is more predictable and deps update branches are more explicitly named.

> It would be useful if it would be possible to use multiple actions pointing into different branches

Yes, I suppose we don't plan anything like that for now.